src/appleseed/renderer/modeling/postprocessingeffect/postprocessingeffectjob.cpp

```diff
@@ -0,0 +1,120 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2020 Tiago Chaves, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "postprocessingeffectjob.h"
+
+// appleseed.renderer headers.
+#include "renderer/modeling/frame/frame.h"
+
+// appleseed.foundation headers.
+#include "foundation/image/canvasproperties.h"
+#include "foundation/image/image.h"
+#include "foundation/image/tile.h"
+#include "foundation/math/ordering.h"
+
+// Standard headers.
+#include <cassert>
+
+using namespace foundation;
+
+namespace renderer
+{
+
+//
+// EffectJob class implementation.
+//
+
+EffectJob::EffectJob(
+    const EffectApplierVector&  effect_appliers,
+    const Frame&                frame,
+    const size_t                tile_x,
+    const size_t                tile_y,
+    const size_t                thread_count)
+  : m_effect_appliers(effect_appliers)
```

Comment:
We should name this vector the same way we do in other job renderers. Something like `m_jobs`  ? It makes more sense then when you call `execute()` on a job.

Reply:
I think this is correct, i.e., the job has a "list of appliers" and the job factory that has a "list of jobs".
Like the [TileJob](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed/renderer/kernel/rendering/generic/tilejob.h#L64) which has a "list of renderers" but no job list. It's the [TileJobFactory](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed/renderer/kernel/rendering/generic/tilejobfactory.h#L80) that has a `tile_jobs`.